server/src/main/java/org/elasticsearch/cluster/version/VersionsWrapper.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.version;
+
+import org.elasticsearch.TransportVersion;
+
+import java.util.Comparator;
+import java.util.Map;
+
+/**
+ * Wraps component version numbers for cluster state
+ *
+ * <p>Cluster state will need to carry version information for different independently versioned components.
+ * This wrapper lets us wrap these versions one level below {@link org.elasticsearch.cluster.ClusterState}.
+ * It's similar to {@link org.elasticsearch.cluster.node.VersionInformation}, but this class is meant to
+ * be constructed during node startup and hold values from plugins as well.
+ *
+ * @param transportVersion
+ */
+public record VersionsWrapper(TransportVersion transportVersion) {
```

Comment:
Could we name this a bit more declaratively? Something like `ClusterVersions`?

Reply:
I'll definitely change the name, but I'm going to save that for the last step of the PR so that I'm not changing terms in the middle of github discussions.